server/internal/auth/chatsessions/jwt.go

```diff
@@ -11,35 +11,29 @@ import (
 
 // ChatSessionClaims represents the custom claims for a chat session JWT
 type ChatSessionClaims struct {
-	OrgID            string  `json:"org_id"`
-	ProjectID        string  `json:"project_id"`
-	OrganizationSlug string  `json:"organization_slug"`
-	ProjectSlug      string  `json:"project_slug"`
-	UserIdentifier   *string `json:"user_identifier,omitempty"`
+	OrgID            string   `json:"org_id"`
+	ProjectID        string   `json:"project_id"`
+	OrganizationSlug string   `json:"organization_slug"`
+	ProjectSlug      string   `json:"project_slug"`
+	UserIdentifier   *string  `json:"user_identifier,omitempty"`
+	AllowedOrigins   []string `json:"allowed_origins,omitempty"`
```

Comment:
The token should be issued for a given origin and not reused across all allowed origins. I think this should a `string` and `[]string` and could utilize the existing `aud` JWT claim.